src/DotNetWorker.ApplicationInsights/FunctionsApplicationInsightsExtensions.cs

```diff
@@ -3,65 +3,27 @@
 
 using System;
 using Microsoft.ApplicationInsights.Extensibility;
-using Microsoft.ApplicationInsights.WorkerService;
 using Microsoft.Azure.Functions.Worker.ApplicationInsights;
 using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.DependencyInjection.Extensions;
-using Microsoft.Extensions.Logging;
-using Microsoft.Extensions.Logging.ApplicationInsights;
 
 namespace Microsoft.Azure.Functions.Worker
 {
     public static class FunctionsApplicationInsightsExtensions
     {
-        /// <summary>
-        /// Adds Application Insights support by internally calling <see cref="ApplicationInsightsExtensions.AddApplicationInsightsTelemetryWorkerService(IServiceCollection)"/>.
-        /// </summary>
-        /// <param name="builder">The <see cref="IFunctionsWorkerApplicationBuilder"/></param>
-        /// <param name="configureOptions">Action to configure ApplicationInsights services.</param>
-        /// <returns>The <see cref="IFunctionsWorkerApplicationBuilder"/></returns>
-        public static IFunctionsWorkerApplicationBuilder AddApplicationInsights(this IFunctionsWorkerApplicationBuilder builder, Action<ApplicationInsightsServiceOptions>? configureOptions = null)
+        public static IServiceCollection ConfigureFunctionsApplicationInsights(this IServiceCollection services)
```

Comment:
Didn't we also talk about some PostConfigureOptions call that would throw if an ITelemetryConfig wasn't registered with DI?

I don't recall the exact mechanism you'd mentioned... but just some way to make sure that the customer hasn't *only* called this API and expects it to work.

Reply:
Ah right, will need to think about that a bit more. The issue with the `PostConfigure` is that something needs to resolve the imports for that validation to happen. So need a reliable startup hook that _isn't_ dependent on AppInsights being present.

There is a host startup, which may work.

Reply:
Added!